tests/integration/models/device.spec.coffee

```diff
@@ -753,6 +759,40 @@ describe 'Device Model', ->
 				.then ({ is_accessible_by_support_until__date }) ->
 					m.chai.expect(is_accessible_by_support_until__date).to.be.null
 
+		describe 'resin.models.device.tags', ->
+
+			appTagTestOptions =
+				model: resin.models.device.tags
+				resourceName: 'application'
+				uniquePropertyName: 'app_name'
+
+			deviceTagTestOptions =
+				model: resin.models.device.tags
+				resourceName: 'device'
+				uniquePropertyName: 'uuid'
+
+			beforeEach ->
+				appTagTestOptions.resource = @application
+				deviceTagTestOptions.resource = @device
+				# used for tag creation during the
+				# device.tags.getAllByApplication() test
+				appTagTestOptions.setTagResource = @device
+
+			describe 'resin.models.device.tags.getAllByApplication()', ->
+				itShouldGetAllTagsByResource(appTagTestOptions)
```

Comment:
Since `appTagTestOptions` only gets used once, here, I'd just inline it

Reply:
Couldn't make it work since `describe` is called before the `beforeEach` where the `@device` becomes available.
Also tried using getter methods, but since the `this` inside `describe` and `beforeEach` are different, I still needed a higher level object to reference in both of them, so I couldn't inline it.
Would you prefer to make `itShouldGetAllTagsByResource` (and the rest methods) to expect the existence of `@resource` on their scope instead of piggy bugging in the options object?

Reply:
:+1: yep, this is also me forgetting that we need to wait before `@device` is available. I definitely think passing options is clearer than having an implicit `this` property, let's just leave this as is.